pkg/apis/build/v1alpha1/build_pod.go

```diff
@@ -390,12 +391,17 @@ func (b *Build) BuildPod(images BuildPodImages, secrets []corev1.Secret, taints
 							"-project-metadata=/layers/project-metadata.toml"},
 							cacheArgs,
 							func() []string {
-								if platformAPI == "0.3" {
+								platformAPIVersion := semver.MustParse(platformAPI)
```

Comment:
A bit concerned about the "MustParse" here a bad builder with an invalid metadata could lead to a panic.